### ISSUE TITLE
Can't work with moleculer-decorators

### ISSUE THREAD

**Issue by pewh** [dkuida/moleculer-db-adapter-typeorm#10]
```
@Service({
	name: 'posts',
	mixins: [storeService],
    adapter: new TypeOrmDbAdapter({
        database: 'memory',
        name: 'memory',
        type: 'sqlite',
    }),

    model: Post,
    settings: {
        fields: ['id', 'title', 'content', 'votes', 'status', 'author'],
        idField: 'id'
    },
    actions: {
        vote(ctx: Context) {
            return this.adapter.findById(ctx.params.id)
                .then((post: Post) => {
                    post.votes++;
                    return this.adapter.repository.save(post);
                })
                .then(() => this.adapter.findById(ctx.params.id))
                .then((doc: any) => this.transformDocuments(ctx, ctx.params, doc));
        },

        unvote(ctx: Context) {
            return this.adapter.findById(ctx.params.id)
                .then((post: Post) => {
                    post.votes--;
                    return this.adapter.repository.save(post);
                })
                .then(() => this.adapter.findById(ctx.params.id))
                .then((doc: any) => this.transformDocuments(ctx, ctx.params, doc));
        }
    },
    afterConnected() {
        this.logger.info('Connected successfully');
        return this.adapter.clear();
    }
})
class tes {   
}

broker.createService(tes)

...
```

throw this error:

```
/home/pewh/Downloads/moleculer-db-adapter-typeorm-develop/node_modules/reflect-metadata/Reflect.js:544
                        throw new TypeError();
                        ^

TypeError
    at DecorateConstructor (/home/pewh/Downloads/moleculer-db-adapter-typeorm-develop/node_modules/reflect-metadata/Reflect.js:544:31)
    at Object.decorate (/home/pewh/Downloads/moleculer-db-adapter-typeorm-develop/node_modules/reflect-metadata/Reflect.js:130:24)
    at __decorate (/home/pewh/Downloads/moleculer-db-adapter-typeorm-develop/dist/examples/integration/index.js:4:92)
    at Object.<anonymous> (/home/pewh/Downloads/moleculer-db-adapter-typeorm-develop/dist/examples/integration/index.js:26:7)
    at Module._compile (internal/modules/cjs/loader.js:707:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:718:10)
    at Module.load (internal/modules/cjs/loader.js:605:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:544:12)
    at Function.Module._load (internal/modules/cjs/loader.js:536:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:760:12)
    at startup (internal/bootstrap/node.js:303:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:872:3)

```

**Comment by dkuida** [comment]
Sorry for the delay, could you check that you have `reflect-metadata` installed as it is a pre request of typeorm ? 
As well I notice that your class does not inherit from  `extends BaseSchema` as instructed in the decorators [doc](https://github.com/ColonelBundy/moleculer-decorators#readme) 

your code should look like that 

```
import {Action, BaseSchema, Method, Service} from 'moleculer-decorators';
import {Context} from 'moleculer';
@Service({
    name: 'posts',
    mixins: [storeService],
    adapter: new TypeOrmDbAdapter({
        database: 'memory',
        name: 'memory',
        type: 'sqlite',
    }),

    model: Post,
    settings: {
        fields: ['id', 'title', 'content', 'votes', 'status', 'author'],
        idField: 'id'
    },

    
})
class tes extends BaseSchema{

    @Action()
    public  vote(ctx: Context) {
        return this.adapter.findById(ctx.params.id)
            .then((post: Post) => {
                post.votes++;
                return this.adapter.repository.save(post);
            })
            .then(() => this.adapter.findById(ctx.params.id))
            .then((doc: any) => this.transformDocuments(ctx, ctx.params, doc));
    }

    @Action()
    public unvote(ctx: Context) {
        return this.adapter.findById(ctx.params.id)
            .then((post: Post) => {
                post.votes--;
                return this.adapter.repository.save(post);
            })
            .then(() => this.adapter.findById(ctx.params.id))
            .then((doc: any) => this.transformDocuments(ctx, ctx.params, doc));
    }
    @Method
    afterConnected() {
        this.logger.info('Connected successfully');
        return this.adapter.clear();
    }

}

module.exports = tes;


broker.createService(tes)

```

**Comment by dkuida** [comment]
So after some checking it turns out the issue is in the way "reflect-metadata" checks for constructor
[            return typeof argument === "function";](https://github.com/rbuckton/reflect-metadata/blob/c2dbe1d02ceb9987f9002eedf0cdb21d74de0019/Reflect.ts#L1494) 
which is flawed, and should check for || argument.constructor.
I will check if I could create a pull request for them

**Comment by dkuida** [comment]
@pewh  solved with version #0.1.2
